File: internal/commands/adr.go
package commands

import (
	"fmt"
	"github.com/kgsaran/trackfw/internal/homedir"
	"os"

	"github.com/charmbracelet/huh"
	cbterm "github.com/charmbracelet/x/term"
	"github.com/kgsaran/trackfw/internal/config"
	"github.com/kgsaran/trackfw/internal/generators"
	"github.com/spf13/cobra"
)

func newADRCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "adr",
		Short: "Manage Architecture Decision Records",
	}
	cmd.AddCommand(newADRNewCmd())
	cmd.AddCommand(newADRListCmd())
	return cmd
}

func newADRNewCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "new <title>",
		Short: "Create a new ADR",
		Args:  cobra.ExactArgs(1),
		RunE: func(cmd *cobra.Command, args []string) error {
			scope, err := cmd.Flags().GetString("scope")
			if err != nil {
				return err
			}
			adrDir, err := resolveADRDir(scope)
			if err != nil {
				return err
			}

			content := generators.ADRContent{Title: args[0]}

			// Detectar se stdin é TTY — wizard interativo somente em TTY
			if cbterm.IsTerminal(uintptr(os.Stdin.Fd())) {
				form := huh.NewForm(
					huh.NewGroup(
						huh.NewInput().
							Title("Context").
							Description("What is the situation that motivates this decision?").
							Value(&content.Context),
					),
					huh.NewGroup(
						huh.NewInput().
							Title("Decision").
							Description("What was decided?").
							Value(&content.Decision),
					),
					huh.NewGroup(
						huh.NewInput().
							Title("Consequences").
							Description("What are the positive and negative consequences?").
							Value(&content.Consequences),
					),
					huh.NewGroup(
						huh.NewInput().
							Title("Alternatives Considered").
							Description("What other options were evaluated and why were they rejected?").
							Value(&content.Alternatives),
					),
				)
				if err := form.Run(); err != nil {
					return fmt.Errorf("wizard: %w", err)
				}
			}

			return generators.NewADR(content, adrDir)
		},
	}
	cmd.Flags().String("scope", "project", "ADR scope: project (docs/adr, default) or global (~/.trackfw/adr, cross-project)")
	return cmd
}

func newADRListCmd() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "list",
		Short: "List all ADRs in docs/adr/",
		RunE: func(cmd *cobra.Command, args []string) error {
			scope, err := cmd.Flags().GetString("scope")
			if err != nil {
				return err
			}
			adrDir, err := resolveADRDir(scope)
			if err != nil {
				return err
			}
			return generators.ListADRs(adrDir)
		},
	}
	cmd.Flags().String("scope", "project", "ADR scope: project (docs/adr, default) or global (~/.trackfw/adr, cross-project)")
	return cmd
}

// resolveADRDir resolves the ADR directory to use based on --scope. "project" (default)
// reads adr_dirs[0] from trackfw.yaml, same as before this flag existed. "global" resolves
// to ~/.trackfw/adr without requiring trackfw.yaml/a project root in the cwd — same pattern
// as UpdateHarness, which never requires a project.
func resolveADRDir(scope string) (string, error) {
	switch scope {
	case "project", "":
		return config.Load().ADRDirs[0], nil
	case "global":
		home, err := homedir.Dir()
		if err != nil {
			return "", fmt.Errorf("localizando home dir: %w", err)
		}
		return generators.GlobalADRDir(home), nil
	default:
		return "", fmt.Errorf("--scope inválido: %q (use \"project\" ou \"global\")", scope)
	}
}

File: npm/src/commands/adr.js
'use strict'

const path = require('path')
const os = require('os')
const { Command } = require('commander')
const { input } = require('@inquirer/prompts')
const generators = require('../generators/adr')
const { t } = require('../i18n')
const { homedir } = require('../homedir')

const cmd = new Command('adr')
cmd.description(t('adr.description'))

/**
 * Resolve o diretório de ADRs conforme --scope.
 * 'project' (default) preserva o comportamento atual (adrDirs[0] do trackfw.yaml).
 * 'global' escreve/lista em ~/.trackfw/adr, sem exigir trackfw.yaml no cwd.
 * @param {string} scope
 * @returns {string}
 */
function resolveAdrDir(scope) {
  if (scope === 'global') {
    return path.join(homedir(), '.trackfw', 'adr')
  }
  if (scope === 'project') {
    return require('../config').load().adrDirs[0]
  }
  console.error(`Invalid --scope value: "${scope}". Expected "project" or "global".`)
  process.exit(1)
}

cmd.command('new <title>')
  .description(t('adr.new.description'))
  .option('--scope <scope>', 'ADR scope: project (default) or global (~/.trackfw/adr)', 'project')
  .action(async (title, options) => {
    const adrDir = resolveAdrDir(options.scope)
    const content = { title }
    // wizard interativo se TTY
    if (process.stdin.isTTY) {
      content.context = await input({ message: t('adr.new.prompt.context'), default: '' })
      content.decision = await input({ message: t('adr.new.prompt.decision'), default: '' })
      content.consequences = await input({ message: t('adr.new.prompt.consequences'), default: '' })
      content.alternatives = await input({ message: t('adr.new.prompt.alternatives'), default: '' })
    }
    await generators.newADR(content, adrDir)
  })

cmd.command('list')
  .description(t('adr.list.description'))
  .option('--scope <scope>', 'ADR scope: project (default) or global (~/.trackfw/adr)', 'project')
  .action(async (options) => {
    const adrDir = resolveAdrDir(options.scope)
    await generators.listADRs(adrDir)
  })

module.exports = cmd
